FirebaseAppCheck/Tests/Unit/Core/FIRAppCheckTests.m

```diff
@@ -304,7 +218,8 @@ - (void)testGetToken_AppCheckProviderError {
                }];
 
   // 3. Wait for expectations and validate mocks.
-  [self waitForExpectations:expectations timeout:0.5];
```

Comment:
This test and `testInteropGetTokenForcingRefresh_Success` have been flaky (both locally and on CI). My hypothesis is that this is due to the inverted tokenNotification expectation which must wait the full duration of the timeout (since it's testing that a notification was _not_ posted).

I've split these into two waits since we want to ensure that the `tokenForcingRefresh` completion handler is invoked and then give a little time to ensure that no notification is posted after. Will see if this addresses the flakiness.

Reply:
Didn't resolve the issue. Replaced `tokenUpdateNotificationWithExpectedToken:isInverted:` with a new `tokenUpdateNotificationNotPosted` to handle the `isInverted = YES` case.

Reply:
Can you disable xcpretty for just these tests and spit the full xcodebuild logs into GHA? It's short enough that it's probably fine and it may contain a more complete failure reason. Right now it just [says test failed](https://github.com/firebase/firebase-ios-sdk/actions/runs/6254122844/job/16980998838?pr=11721) but doesn't say why.

Reply:
Done (and added back now). Thanks for the debugging help!